### DISCUSSION TITLE
Can I "poke" the output leds from basic ?

### DISCUSSION THREAD

**Discussion by ghost** [thehighnibble/thehighnibble.github.io#73] [IMSAI8080esp replica]
Hey everyone,

I tried to output data to the output leds from basic inside cp/m but that do'esnt seem to work
Am I doing something wring ? Maybe cp/m prevents something ?
What i/o port should I use, maybe I'm doing it wrong

thanks in advance,

**Comment by agn453** [comment]
The programmable output display register is at I/O address 0FFh (255 decimal).

It displays the complement value of what is written to it, therefore to set all
leds ON you do an ```OUT 255,0``` in MBASIC. (write zero to I/O port address 255).

Tony

**Comment by GrantMeStrength** [comment]
What were you used from CP/M to access the I/O ports?
I've been able to control them with BASIC and Turbo Pascal.
I bet assembler would work too.

**Comment by omiq** [comment]
Thanks for sharing that, here is my hello world equiv

```
10 I=255
20 OUT 255,I
30 FOR I=255 TO 0 STEP-1
40 OUT 255,I
50 FOR W=1 TO 100:NEXT W
60 NEXT I
70 GOTO 10
```
